src/transport.ts

```diff
@@ -33,7 +33,31 @@ export async function startStdioTransport(config: FullConfig, connectionList: Co
   connectionList.push(connection);
 }
 
+function checkCors(config: FullConfig, req: http.IncomingMessage, res: http.ServerResponse): boolean {
+  if (config.network?.corsAllowOrigins === undefined)
+    return false;
+
+  const origin = req.headers.origin;
+  if (origin === undefined)
+    return false;
+
+  if (config.network.corsAllowOrigins.some(re => re.test(origin))) {
```

Comment:
What is your use case? This exposes universal access to the host machine to the client, including full access to the fs.